packages/manager/apps/pci-object-storage/src/api/hooks/useStorages.ts

```diff
@@ -547,3 +575,74 @@ export const useStorageEndpoint = (projectId: string, storage: TStorage) => {
         }/${storage?.name}`,
   };
 };
+
+export const useAllServerStorages = (
+  projectId: string,
+  availability: {
+    isLocalZoneAvailable: boolean;
+    is3azAvailable: boolean;
+  },
+) => {
+  const { i18n, t } = useTranslation(['containers', 'containers/add']);
+  const queryClient = useQueryClient();
+
+  const { data, isLoading, isPending, isRefreshing, error } = useMappedStorages(
+    projectId,
+  );
+
+  const mappedStorages = useMemo(() => {
```

Comment:
We can consider refactoring to 
- decouple responsibilities
- prevent from mutability with `mode` variable.
- handle the `default` case in the switch